src/HazelcastClient.ts

```diff
@@ -44,8 +44,8 @@ class HazelcastClient {
             return this.partitionService.initialize();
         }).then(() => {
             deferred.resolve(this);
-        }).catch(() => {
-            deferred.reject('Client failed to start');
+        }).catch((e) => {
+            deferred.reject(e);
```

Comment:
Before the change in case of the error you would only get a failure message without any explanation.